### PR TITLE
Feature/inject init containers

### PR DESCRIPTION
**Pull request by iwilltry42**
# What and why?

*Note*: this PR assumes that the [hostAliases PR](https://github.com/tumblr/k8s-sidecar-injector/pull/5) got merged already (too lazy for some more merges).

*Note 2*: I "simplified" adding tests in `internal/pkg/config/config_test.go` similar to how it's done in `loader_test.go`.

Functionality to add initContainers, because they're quite useful sometimes and some functionality, that we currently execute in sidecar containers may only be needed once at the start of a pod.

This works similar to the injection of "normal" containers, meaning that, if there's a name conflict between injection config and the requesting podSpec, the server will return a `Duplicate Value` error.

# Testing Steps

Tested with debug-pod that requests an injection with one initContainer in it.
Configs of debug-pod tested:
- 0 existing initContainers
- 1 existing initContainer
- 1 existing initContainer with name conflict 

Same unittests as usual.

# Reviewers

Required reviewers: `@byxorna`
Request reviews from other people you want to review this PR in the "Reviewers" section on the right.

### CODE REVIEW

File: internal/pkg/config/config_test.go [hunk 1]
```diff
@@ -6,17 +6,127 @@ import (
...
-	sidecars          = "test/fixtures/sidecars"
-	cfg1              = sidecars + "/sidecar-test.yaml"
-	complicatedConfig = sidecars + "/complex-sidecar.yaml"
-	env1              = sidecars + "/env1.yaml"
-	volumeMounts      = sidecars + "/volume-mounts.yaml"
+	// location of the fixture sidecar files
+	fixtureSidecarsDir = "test/fixtures/sidecars"
+
+	// test files and expectations
+	testConfigs = map[string]configExpectation{
```
Comment (byxorna) [inline internal/pkg/config/config_test.go]: thank you, this is MUCH nicer

**Comment by iwilltry42** [comment]
@zach-dunton-sf can I merge your changes regarding annotations into this PR? That way we can have all the functionality without having to mess with merge conflicts. And as far as I can see, our changes regarding the injection of initContainers are fairly similar.

**Comment by iwilltry42** [comment]
@byxorna I pasted @zach-dunton-sf 's changes regarding annotations into my branch :+1:

File: docs/sidecar-configuration-format.md [hunk 2]
```diff
@@ -59,6 +59,13 @@ env:
 volumeMounts:
   - name: some-config
     mountPath: /etc/some-config
+
+# initContainers will be added, no replacement of existing initContainers with the same names will be done
+# this works exactly the same way like adding normal containers does: if you have a conflicting name, the server will return an error
```
Comment (komapa) [inline docs/sidecar-configuration-format.md]: I suggest breaking `the server will return an error` on a new line

File: test/fixtures/k8s/object7.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+name: object6
```
Comment (komapa) [inline test/fixtures/k8s/object7.yaml]: `name: object7`

Comment (iwilltry42) [reply test/fixtures/k8s/object7.yaml]: Oops... changed that :+1:

File: docs/sidecar-configuration-format.md (hunk 2 above)
Comment (iwilltry42) [reply docs/sidecar-configuration-format.md]: Done

**Comment by byxorna** [comment]
@iwilltry42 ok, thanks for all this! I am merging and will cut a new release in a bit; you can use the `:latest` image for now, until I write up the release notes :)

**Pull request merged**